sophys_gui/components/form/main.py

```diff
@@ -467,6 +497,76 @@ def updateParametersLayout(self, newGroup):
         self.parametersLayout.addWidget(newGroup)
         self.group = newGroup
 
+    def addInputWidget(self, paramMeta, pos, glay):
+        pos = self.addParameterInput(paramMeta, pos, glay)
+        if self.readingOrder == "up_down":
+            if pos[0] >= self.max_rows*2:
+                pos[0] = 0
+                pos[1] += 2
+        else:
+            if pos[1] >= self.max_cols - 1:
+                pos[1] = 0
+                pos[0] += 1
+            else:
+                pos[1] += 1
+                pos[0] -= 2
+
+    def addParameters(self, parameters, glay):
+        pos = [0, 0]
+        for paramMeta in parameters:
+            self.addInputWidget(paramMeta, pos, glay)
+
+    def changePlanName(self):
+
+        if self.yml_file_path:
+            plan_config = self.config.get("plans", {}).get(self.chosenItem)
```

Comment:
Could you remove the beamline and plans in the YML? As they are not used in the GUI, then you can remove the lines to get the plans, the are some like this one that could be simplified.